### ISSUE TITLE
add libraries that are in included in the "program files" path

### ISSUE THREAD

**Issue by eduardotsimoes** [conan-io/conan#5382]
hello, 
First of all, I am using the conan version 1.11.1 in windows 10.

I am creating a binary package that needs some intel fortran libraries to compile. Those libraries can be installed in the "program files" folder. So, to do this I implemented this way the package_info() method:

```
    def package_info(self):
        if self.settings.os == "Linux":
            self.cpp_info.libs = ["libbla.a", "gfortran", "quadmath"]
        if self.settings.os == "Windows":
            fortranLibsPath = os.path.join(os.environ["IFORT_COMPILER19"],  "compiler\\lib\\intel64_win\\")
            libs= ["ifconsol.lib","libirc.lib","libifcoremd.lib","libifportmd.lib","libmmd.lib","svml_dispmd.lib"]
            self.cpp_info.libs = ["bla.lib"]
            for lib in libs:
                LibPath = "\""+ os.path.join(fortranLibsPath,lib)+"\""
                self.cpp_info.libs.append(LibPath) 
```
The problem is that the double quotes are changed by "\\"" in:
https://github.com/conan-io/conan/blob/8e4fe93a984d5cb0d81596b44e634d2f42ae8700/conans/client/generators/cmake.py#L18

and when the file is saved by 
https://github.com/conan-io/conan/blob/c4470a6dbfe3d80b911711215264d53310d40cbd/conans/client/generators/__init__.py#L110

The \\" becomes a \" instead of a simple ". 

Is there a better way to deal with path with spaces between?

To solve this, I am calling the DOS to give the short name,  but this is not elegant in anyway.

```
    def package_info(self):
        if self.settings.os == "Linux":
            self.cpp_info.libs = ["libBla.a", "gfortran", "quadmath"]
        if self.settings.os == "Windows":
            fortranLibsPath = os.path.join(os.environ["IFORT_COMPILER19"],  "compiler\\lib\\intel64_win\\")
            for_cmd = 'for %I in ("' + fortranLibsPath + '") do echo %~sI'
            p = os.popen(for_cmd)
            short_name = p.readlines()[-1].replace("\n","")
            libs= ["ifconsol.lib","libirc.lib","libifcoremd.lib","libifportmd.lib","libmmd.lib","svml_dispmd.lib"]
            self.cpp_info.libs = ["bla.lib"]
            for lib in libs:
                LibPath =  os.path.join(short_name,lib)
                self.cpp_info.libs.append(LibPath) 
```

Thanks in advance,
Eduardo

**Comment by lasote** [comment]
Probably you can use better "/" instead of "\\". Windows works usually nice with it:

```
fort_compiler_path = os.environ["IFORT_COMPILER19"].replace("\\", "/"),
fortranLibsPath = "%s/compiler/lib/intel64_win/" % (fort_compiler_path)
         
```

**Comment by Johnnyxy** [comment]
@eduardotsimoes 
Hi, 
first of all, you do not need to enclose the paths in quotes. Python as well als CMake can handle spaces in paths and enclose them in quotes automatically if necessary. Let the Conan generators and CMake find the correct libraries:
```python
def package_info(self):
    self.cpp_info.libdirs = [os.path.join(os.environ["IFORT_COMPILER19"],  "compiler/lib/intel64_win")]
    self.cpp_info.libs = ["bla.lib", "ifconsol.lib","libirc.lib","libifcoremd.lib","libifportmd.lib","libmmd.lib","svml_dispmd.lib"]
```

But I do not know if your compiler supports spaces in paths or quotes around paths. E.g. some GCC Windows ports do not. Thus you cannot invoke them with spaces in the paths of the libs.
If your compiler does not support spaces in paths then you need the short file names. In such cases the following should work (your approach modified):
```python
def package_info(self):
    self.cpp_info.libs = ["bla.lib"]
    libs = ["ifconsol.lib","libirc.lib","libifcoremd.lib","libifportmd.lib","libmmd.lib","svml_dispmd.lib"]
    libfilenames = [os.path.join(fortranLibsPath, lib) for lib in libs]
    for f in libfilenames:
        fortranLibsPath = os.path.join(os.environ["IFORT_COMPILER19"],  "compiler/lib/intel64_win")
        res = os.popen('for %I in ("' + fortranLibsPath + '") do echo %~sI')
        short_name = p.readlines()[-1].replace("\n","")
        self.cpp_info.libs += [short_name]
```
This way you do get the short path of each file.

**Comment by eduardotsimoes** [comment]
Hello, 

thanks a lot both for the great help. 

I was looking for this kind of clean solution
```
def package_info(self):
    self.cpp_info.libdirs = [os.path.join(os.environ["IFORT_COMPILER19"],  "compiler/lib/intel64_win")]
    self.cpp_info.libs = ["bla.lib", "ifconsol.lib","libirc.lib","libifcoremd.lib","libifportmd.lib","libmmd.lib","svml_dispmd.lib"]
```

but in this case, I need two libdirs. One is from the compiler libraries and the other is from the library itself that was compiled in fortran.   When I try your solution, the variable 

set(CONAN_INCLUDE_DIRS_BLA "c:\fortranlibs\")

in cmake just have the fortran folder, the original one is lost. 

I tried the following:

`self.cpp_info.libdirs = [os.path.join(os.environ["IFORT_COMPILER19"],  "compiler/lib/intel64_win"), os.path.join(self.cpp_info.lib_paths[0]) ] `

But it removes the first path.  

Looking in the souce code, I found that the variable was set here:
https://github.com/conan-io/conan/blob/d815f1da95a9acd2dabd7743a9ac81b139e0edb6/conans/client/generators/cmake_common.py#L3

using the variable deps.lib_paths that was defined here :

https://github.com/conan-io/conan/blob/8e4fe93a984d5cb0d81596b44e634d2f42ae8700/conans/client/generators/cmake.py#L28

so, I tried this modification : 

```
    def package_info(self):
        if self.settings.os == "Linux":
            self.cpp_info.libs = ["libBla.a", "gfortran", "quadmath"]
        if self.settings.os == "Windows":
            self.cpp_info.lib_paths.append(os.path.join(os.environ["IFORT_COMPILER19"],  "compiler/lib/intel64_win"))
            self.cpp_info.libs = ["Bla.lib", "ifconsol.lib","libirc.lib","libifcoremd.lib","libifportmd.lib","libmmd.lib","svml_dispmd.lib"]
```

And it works perfect. But I am not sure if it is safe to use the lib_paths variable. Is it OK? Is it the best practice in conan?
